### PR TITLE
Feature/required info description

### PR DESCRIPTION
**Pull request by markhelmus**
Note that app crashes if this.serverCollectionsInteractor.getHandler().getIp() is null (in IpDialog, line 32)

### CODE REVIEW

File: Hestia/app/src/main/java/hestia/UI/dialogs/AddDeviceInfo.java [hunk 1]
```diff
@@ -52,16 +58,36 @@ protected void onCreate(Bundle savedInstanceState) {
...
             name.setText(key);
+            name.setOnLongClickListener(new View.OnLongClickListener() {
+                @Override
+                public boolean onLongClick(View v) {
+                    AlertDialog.Builder builder = new AlertDialog.Builder(context);
+
+                    builder
+                            .setMessage(
+                                    fields.get(key));
+                    AlertDialog dialog = builder.create();
```
Comment (StefanEvanghelides) [inline Hestia/app/src/main/java/hestia/UI/dialogs/AddDeviceInfo.java]: not very important, but we could rearrange this part

File: Hestia/app/src/main/java/hestia/UI/dialogs/HestiaDialog.java [hunk 2]
```diff
@@ -10,48 +12,70 @@
...
+//        TextView header = (TextView)findViewById(R.id.dialog_header);
+////        header.setText(title);
+////        cancel = (Button)findViewById(R.id.cancel_button);
+////        cancel.setOnClickListener(this);
+////        confirm = (Button)findViewById(R.id.confirm_button);
+////        confirm.setOnClickListener(this);
+//        LayoutInflater inflater = getLayoutInflater();
+//
+//    }
+
```
Comment (StefanEvanghelides) [inline Hestia/app/src/main/java/hestia/UI/dialogs/HestiaDialog.java]: Is this part still needed? If not, remove it

File: Hestia/app/src/main/java/hestia/UI/dialogs/HestiaDialog.java [hunk 3]
```diff
@@ -10,48 +12,70 @@
...
     abstract void pressCancel();
 
     abstract void pressConfirm();
 
     public void onClick(View view) {
         switch (view.getId()) {
             case R.id.cancel_button:
-                pressCancel();
-                dismiss();
+
```
Comment (StefanEvanghelides) [inline Hestia/app/src/main/java/hestia/UI/dialogs/HestiaDialog.java]: You don't need this switch statement anymore, do you?

File: Hestia/app/src/main/java/hestia/UI/dialogs/IpDialog.java [hunk 4]
```diff
@@ -49,4 +51,9 @@ void pressConfirm() {
                     Toast.LENGTH_SHORT).show();
         }
     }
+
```
Comment (StefanEvanghelides) [inline Hestia/app/src/main/java/hestia/UI/dialogs/IpDialog.java]: I suggest we create a new method that checks the entered ip address. Such a check would include the following:
- ip is not null, nor is it empty.
- should contain only numbers.
- should have the correct formating of an ip (v4/v6, groups of numbers not greater than 255)

File: Hestia/app/src/main/java/hestia/UI/dialogs/HestiaDialog.java [hunk 5]
```diff
@@ -10,48 +12,70 @@
...
+                dismiss();
+            }
+        });
         LayoutInflater inflater = getLayoutInflater();
-        FrameLayout frame = (FrameLayout)findViewById(R.id.dialog_container);
+//        FrameLayout frame = (FrameLayout)findViewById(R.id.dialog_container);
         View view = inflater.inflate(layoutReference, null);
-        frame.addView(view);
+//        frame.addView(view);
+        //TODO: better code
```
Comment (StefanEvanghelides) [inline Hestia/app/src/main/java/hestia/UI/dialogs/HestiaDialog.java]: Finish the TODO first, then remove it

**Review by StefanEvanghelides — CHANGES_REQUESTED** [review]
Interesting change from Dialog to AlertDialog. I added separate comments to parts that could be improved.
Don' forget to add comments to these clases

File: Hestia/app/src/main/java/hestia/UI/dialogs/IpDialog.java (hunk 4 above)
Comment (Feikoritsema) [reply Hestia/app/src/main/java/hestia/UI/dialogs/IpDialog.java]: Check my ip branch pull request for How to use Java Patterns (regex) to check this. If I have the time today i Will add it.

**Comment by StefanEvanghelides** [comment]
Tests still fail; Merge conflics

**Review by LarsHoldijk — CHANGES_REQUESTED** [review]
Need to wait for Mark his changes with the dialog

**Pull request closed without merging**